ckanext/datastore/interfaces.py

```diff
@@ -145,3 +145,25 @@ def datastore_delete(self, context, data_dict, fields_types, query_dict):
         :rtype: dictionary
         '''
         return query_dict
+
+
+class IDatastoreBackend(interfaces.Interface):
+    """Allow custom implementations of datastore backend"""
+    def register_backends(self):
+        """
+        Register classes that inherits from DatastoreBackend.
+
+        Every registered class provides implementations of DatastoreBackend
+        and, depending on `datastore.write_url`, one of them will be used
```

Comment:
Is there any check on `datastore.read_url`? eg could you end up using a different backend for read operations (search) and write ones?